dev/devicelab/lib/framework/framework.dart

```diff
@@ -103,18 +107,18 @@ class _TaskRunner {
       if (taskTimeout != null)
         futureResult = futureResult.timeout(taskTimeout);
 
-      TaskResult result = await futureResult;
+      final TaskResult result = await futureResult;
 
-      section('Checking running Dart$exe processes after task...');
+      section('Checking running Dart$exe and ios-deploy processes after task...');
       final List<RunningProcessInfo> afterRunningDartInstances = await getRunningProcesses(
         processName: 'dart$exe',
       ).toList();
-      for (final RunningProcessInfo info in afterRunningDartInstances) {
-        if (!beforeRunningDartInstances.contains(info)) {
-          print('$info was leaked by this test.');
-          if (result is TaskResultCheckProcesses) {
-            result = TaskResult.failure('This test leaked dart processes');
-          }
+      final Set<RunningProcessInfo> afterRunningIosDeployInstances = await getRunningProcesses(
+        processName: 'ios-deploy',
+      ).toSet();
```

Comment:
There should never be an ios-deploy process running after a test, you don't need to check the before and after as you would with dart.